src/main/java/graphql/Assert.java

```diff
@@ -83,11 +84,18 @@ public static void assertTrue(boolean condition) {
         throw new AssertException("condition expected to be true");
     }
 
-    public static void assertFalse(boolean condition, String format, Object... args) {
+    public static void assertFalse(boolean condition, Supplier<String> msg) {
         if (!condition) {
             return;
         }
-        throw new AssertException(format(format, args));
+        throw new AssertException(msg.get());
+    }
+
+    public static void assertFalse(boolean condition) {
+        if (!condition) {
+            return;
+        }
+        throw new AssertException("condition expected to be false");
     }
```

Comment:
The above was missing but present for assertTrue so I made it

Reply:
assertShouldNeverHappen can be as slow as it likes - its ALWAYS going to assert.

I mean we could make it a supplier based on consistency but....I thought its ok to leave it